### PR TITLE
STM32 ADC: use both sampling time common channels

### PR DESCRIPTION
**Pull request by gautierg-st**
For STM32 series that have two ADC sampling time common channels, only one was used.
This PR adds the support for the second one. The first two different acquisition time values are used for the sequence and all further values must match either of them, otherwise generating an error.

### CODE REVIEW

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -1148,16 +1149,25 @@ static int adc_stm32_check_acq_time(const struct device *dev, uint16_t acq_time)
...
-				  uint8_t acq_time_index)
+static int adc_stm32_sampling_time_setup(const struct device *dev, uint8_t id,
+					 uint16_t acq_time)
 {
 	const struct adc_stm32_cfg *config =
 		(const struct adc_stm32_cfg *)dev->config;
 	ADC_TypeDef *adc = config->base;
+	struct adc_stm32_data *data = dev->data;
+
+
```
Comment (erwango) [inline drivers/adc/adc_stm32.c:1164]: Nit: superfluous lines

File: tests/drivers/adc/adc_api/boards/nucleo_wb55rg.overlay [hunk 2]
```diff
@@ -28,7 +28,7 @@
 		reg = <4>;
 		zephyr,gain = "ADC_GAIN_1";
 		zephyr,reference = "ADC_REF_INTERNAL";
-		zephyr,acquisition-time = <ADC_ACQ_TIME_MAX>;
+		zephyr,acquisition-time = <ADC_ACQ_TIME(ADC_ACQ_TIME_TICKS, 248)>;
```
Comment (erwango) [inline tests/drivers/adc/adc_api/boards/nucleo_wb55rg.overlay:31]: If this setting is made purely for testing a specific case, please add a comment about it.

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/adc/adc_stm32.c (hunk 1 above)
Comment (gautierg-st) [reply drivers/adc/adc_stm32.c]: Done

File: tests/drivers/adc/adc_api/boards/nucleo_wb55rg.overlay (hunk 2 above)
Comment (gautierg-st) [reply tests/drivers/adc/adc_api/boards/nucleo_wb55rg.overlay:32]: Done

**Review by erwango — DISMISSED** [review]

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**